### DISCUSSION TITLE
How to switch from Deploy Keys token to Apps

### DISCUSSION THREAD

**Discussion by piramiday** [community/community#24386] [Apps, API and Webhooks]
<p>hi all, I am having some trouble understanding the basic ideas behind Github Apps.</p>
<p>let’s talk by example.</p>
<p><strong>what I currently do</strong><br>
I have several repos under my account.<br>
for just a single repo, I have added a deploy key with R/W access.<br>
I can handle my repo from any pc using that deploy key, having full access to the repo but without having access to my other repos.<br>
I can commit to the repo and commits will show up as coming from my username.</p>
<p><strong>what I’d like to do</strong><br>
I’d like to switch from Deploy Keys to apps in order to experiment with Github Apps.<br>
I’d like to be able to accomplish the same, that is, push commits from any pc to a single repo, possibly showing this difference by the fact that commits are pushed as a bot (e.g. my-test-bot) and not my own user.<br>
I have some trouble understanding how to accomplish this.<br>
is there an easy guide I could follow, starting with the basics, not with webhooks and runners (which I understand because I routinely use Github Actions but are not what I’d like to do now)?</p>
<p>thanks!</p>

**Comment by nethgato** [comment]
<p>Hi, hi <a class="mention" href="/piramiday">@piramiday</a>!</p>
<p>Happy to be of assistance here, but I might not fully grasp your end goal.</p>
<p>Currently, you have a deploy key which resides on your machines that you develop on.</p>
<p>However you would like to create a GitHub OAuth app, for similar purposes. That is, to manipulate your repositories from a web UI that you’ve crafted, which would utilize not a Deploy Key, but an OAuth key.</p>
<p>You also mentioned Actions, which can be used to apply CI logic to your deployment process, but are unique to GitHub applications.</p>
<p>I suppose the best starting place would be here:</p>
<p><a href="https://docs.github.com/en/developers/apps/building-oauth-apps" class="onebox" target="_blank">https://docs.github.com/en/developers/apps/building-oauth-apps</a></p>
<p>Though, I’d love some clarification on whether or not I fully understood your concern and your goal.</p>

**Comment by piramiday** [comment, marked as answer]
<p>I managed to do what I was trying to do. it’s weird, but it works.</p>
<p>instead of creating a deploy key, I have created a Github App and I used an installation token to be able to interact with the repo as the bot, not as my username.<br>
in such a way issues and PRs created by the bot are nicely marked as being linked to a bot.<br>
commits are showed as being pushed by the bot even though at present Github is not adding a “bot” label on those (but they might in the future).</p>
<p>sorry for the weird question.<br>
you can safely delete this thread since it’s more confusing than helpful for most people.</p>
